aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2014,7 +2017,7 @@ static Tensor _matmul_impl(
                    : tensor1.unsqueeze(0).mm(tensor2).squeeze_(0);
   } else if (dim_tensor1 == 2 && dim_tensor2 == 2) {
     return has_out ? at::mm_out(out, tensor1, tensor2) : tensor1.mm(tensor2);
-  } else if (should_fold(tensor1, tensor2)) {
+  } else if (should_fold(tensor1, tensor2, has_out)) {
```

Comment:
Can we better write
```suggestion
  } else if (!has_out && should_fold(tensor1, tensor2)) {
```
?